airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -0,0 +1,151 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from time import sleep
+from typing import Optional
+
+from airflow.compat.functools import cached_property
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.redshift_data import RedshiftDataHook
+
+
+class RedshiftDataOperator(BaseOperator):
+    """
+    Executes SQL Statements against an Amazon Redshift cluster using Redshift Data
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:RedshiftDataOperator`
+
+    :param cluster_identifier: unique identifier of a cluster
+    :param database: the name of the database
+    :param sql: the SQL statement text to run
+    :param db_user: the database user name
+    :param parameters: the parameters for the SQL statement
+    :param secret_arn: the name or ARN of the secret that enables db access
+    :param statement_name: the name of the SQL statement
+    :param with_event: indicates whether to send an event to EventBridge
+    :param timeout: how long in seconds to wait for a response, if 0 don't wait
```

Comment:
very much not obvious that `timeout` means, also, 'sync'

i would make this boolean (since there's already a timout out in base op you can chop this), and call it `nowait` or something

Reply:
yeah that's fine with me, i don't really like it either :) 
just suggested it cus that's what emerged as consensus in another pr

Reply:
though `wait` alone feels a little vague.   maybe `await_completion` or `await_result` or `sync` is better?  i don't have strong opinion please go with what you think is best, even if that is `wait` of course.

Reply:
Agree. Done